### PR TITLE
Use gettext for translations

### PR DESCRIPTION
**Pull request by dbnicholson**
I'm not sure I'm handling the infrastructure the best way or if the (currently uncommented) `Makefile` approach is the way to go. I dumped out all the currently tracked sites and added any translated data to the PO files.

https://phabricator.endlessm.com/T34536

### CODE REVIEW

File: app-info/Makefile [hunk 1]
```diff
@@ -0,0 +1,48 @@
+NULL :=
```
Comment (pwithnall) [inline app-info/Makefile:1]: New files, such as this Makefile, could do with an SPDX copyright/licensing header

**Review by pwithnall — APPROVED** [review]
This LGTM. One minor nitpicky comment about licensing.

I’m also curious as to why you went with make rather than meson. Familiarity and speed, or the ability to copy the make rules from elsewhere easily? Or does Meson not allow this kind of build to be done easily? Not asking you to port to Meson, just interested!

File: app-info/Makefile (hunk 1 above)
Comment (dbnicholson) [reply app-info/Makefile:1]: :+1:

**Comment by dbnicholson** [comment]
> I’m also curious as to why you went with make rather than meson. Familiarity and speed, or the ability to copy the make rules from elsewhere easily? Or does Meson not allow this kind of build to be done easily? Not asking you to port to Meson, just interested!

I honestly don't know how I'd do it. I'm probably the wrong person to be handling this as I just cobbled together these commands for the first time. Some of it is based on my reading of the [gettext Makefile stub](https://git.savannah.gnu.org/gitweb/?p=gettext.git;a=blob;f=gettext-runtime/po/Makefile.in.in;hb=HEAD). I'm not opposed to using meson, though.

File: app-info/Makefile (hunk 1 above)
Comment (pwithnall) [reply app-info/Makefile:1]: Just go ahead and merge once this nitpick is fixed

**Comment by dbnicholson** [comment]
> The cobbling looks good to me though. I was also curious about how to do this in meson, and my guess would be the [`i18n` module](https://mesonbuild.com/i18n-module.html), which seems to have analogues for everything except the `msginit` block (from a quick look through the manual).

Even the gettext Makefile stub doesn't handle the `msginit` case. It just throws an error and tells you to run it manually. I'm probably being too clever here.

> Anyway, I don’t think it’s necessarily a good use of time to port this module to meson right now, so please take these comments as for intellectual curiosity only.

One the one hand I completely agree it's not the best use of time. On the other hand, using meson's supported methods is probably better than bespoke cobbling. I'm going to merge this for the time being, but I'll probably give a meson conversion a shot. There's very little here right now. There are only a handful of tasks that happen:

* Make a `metainfo.xml.in` file from a URL with `pwa-metainfo-generator`. This is always manual since it needs human editing and the result will be checked in.
* Generate the popular metainfo appended sections for the apps listed in `eos-extra.txt` with `generate-eos-extra-metainfo`. These are also checked in but could be done automatically with a build system.
* Generate the translated `metainfo.xml` files. This is 90% of what's happening in this PR and meson presumably has support for this.
* Generate the `eos-extra.xml` appstream catalog with the `generate-eos-extra-appstream`. This is currently checked in but could be done automatically with a build system.
* Upload the generated appstream catalog, screenshots and thumbnails to S3. This is what the `sync-s3.sh` script does. Presumably this would be a custom target.

**Pull request merged**